chips/nrf52/src/comp.rs

```diff
@@ -0,0 +1,301 @@
+        Hysteresis OFFSET(0) NUMBITS(1)[]
+    ]
+];
+
+pub struct Comp {
+    registers: StaticRef<CompRegisters>,
+    client: OptionalCell<&'static dyn analog_comparator::Client>,
+}
+
+impl Comp {
+    const fn new(registers: StaticRef<CompRegisters>) -> Comp {
+        Comp {
+            registers: registers,
+            client: OptionalCell::empty(),
+        }
+    }
+
+    pub fn set_client(&self, client: &'static dyn analog_comparator::Client) {
+        self.client.set(client);
+    }
+
+    // Returns if comparator was already enabled
+    fn enable(&self) {
+        let regs = &*self.registers;
+        // Checks if it's already enabled
+        // Assumes no one else is writing to comp regs directly
+        if regs.enable.matches_any(Enable::ENABLE::Enabled) {
+            return;
+        }
+
+        // Set mode to Differential
+        // Differential and single ended are pretty much the same,
+        // except single-ended gives more options for input and
+        // uses a ref ladder for hysteresis instead of a set voltage
+        regs.mode
+            .write(Mode::OperatingMode::Differential + Mode::SpeedAndPower::Normal);
+        // VIN+ = Pin 0
+        regs.psel.write(PinSelect::PinSelect::AnalogInput0);
+        // VIN- = Pin 1
+        regs.extrefsel
+            .write(ExternalRefSelect::ExternalRefSelect::AnalogRef1);
+        // Disable hysteresis
+        regs.hyst.write(Hysteresis::Hysteresis::CLEAR);
+
+        regs.enable.write(Enable::ENABLE::Enabled);
+        // start comparator
+        regs.events_ready.set(0);
+        regs.tasks_start.set(1);
+        // wait for comparator to be ready
+        // delay is on order of 3 microseconds so spin wait is OK
+        while regs.events_ready.get() == 0 {}
+    }
+
+    fn disable(&self) {
+        let regs = &*self.registers;
+        // stop comparator
+        regs.tasks_stop.set(1);
+        // completely turn comparator off
+        regs.enable.write(Enable::ENABLE::Disabled);
+    }
+
+    pub fn handle_interrupt(&self) {
+        // HIL only cares about upward crossing interrupts
+        let regs = &*self.registers;
+        // VIN+ crossed VIN-
+        if regs.events_up.get() == 1 {
+            // Clear event
+            regs.events_up.set(0);
+            self.client.map(|client| {
+                // Only one channel
+                client.fired(0);
+            });
+        } else if regs.events_down.get() == 1 {
+            regs.events_down.set(0);
+        }
+    }
+}
+
+impl analog_comparator::AnalogComparator for Comp {
+    type Channel = AcChannel;
+
+    // Only one channel, so this parameter's ignored
+    fn start_comparing(&self, _: &Self::Channel) -> ReturnCode {
+        let regs = &*self.registers;
+        self.enable();
+
+        // Enable only up interrupt (If VIN+ crosses VIN-)
+        regs.inten.write(
+            InterruptEnable::UP::SET, /* + InterruptEnable::DOWN::SET*/
+        );
+
+        return ReturnCode::SUCCESS;
```

Comment:
```suggestion
        ReturnCode::SUCCESS
```

I'm genuinely surprised rustfmt didn't catch this (oh, maybe it will? looks like travis isn't running because there's merge conflicts?), but rust preferred style is to end this simply as the above (there are a couple of instances of this)

Reply:
Fixed.